subprojects/docs/src/docs/release/notes.md

```diff
@@ -78,6 +78,48 @@ The `buildSrc` build of the root project continues to use its own build cache co
 However, the `buildSrc` build of any included build will inherit the build cache configuration from the root build.
 For more on configuring the build cache for the root `buildSrc` build, please see [the Userguide section on this topic](userguide/build_cache.html#buildCacheBuildSrc).
 
+### Richer task inputs
+
+The detection of task input and output annotations for `@Nested` properties now works on the runtime type of the nested bean.
+Earlier Gradle versions used the declared type for detecting annotations.
+Moreover, `Iterable`s can now be annotated with `@Nested` and each element of the iterable will be added as a separate nested input.
```

Comment:
I'd start with the status quo, as it is probably not something people have been thinking about before:

> When analyzing `@Nested` task properties for declared input and output sub-properties Gradle used to only observe the declared type of the property. This meant ignoring any sub-properties declared by a runtime sub-type. Since Gradle 4.5 Gradle uses the type of the actual value instead, and hence can discover all sub-properties declared this way. This allows for a few new tricks.

I'd then add a sub-section for tracking nested iterables, and make "Rich Java compiler arguments" another sub-section.

Reply:
Actually, I think I would start with "Rich Java compiler arguments" as a stand-alone feature, explain how it works with the example, and then go on mentioning that you can introduce your own polymorphic values for other properties, too. This would allow a better ramp-up of knowledge for users, and we'd put the feature interesting to much more people first, and the more niche thingy last.